harness/determined/cli/trial.py

```diff
@@ -149,17 +148,33 @@ def get_with_offset(offset: int) -> bindings.v1GetTrialWorkloadsResponse:
 
 
 def download(args: Namespace) -> None:
-    checkpoint = (
-        Determined(args.master, args.user)
-        .get_trial(args.trial_id)
-        .select_checkpoint(
-            latest=args.latest,
-            best=args.best,
-            uuid=args.uuid,
-            sort_by=args.sort_by,
-            smaller_is_better=args.smaller_is_better,
+    det = Determined(args.master, args.user)
+
+    if [args.latest, args.best, args.uuid].count(True) != 1:
```

Comment:
compare with `determined/common/experimental/trial.py::Trial.download`

Reply:
too bad mypy complains about `sum([args.latest, args.best, args.uuid])`.  I like this `.count(True)` better than wrapping all the bools in `int()`.

Reply:
also, are you basically just copy/pasting the `.select_checkpoint()` verification logic here?

You could move the body of `.select_checkpoint()` into an internal `._select_checkpoint()`, where `.select_checkpoint()` exists only to emit the deprecation warning, and `_select_checkpoint()` is reusable by the CLI.

I like that strategy because it sticks to DRY, and avoids any possibility of diverging code.  When `select_checkpoint` is removed from the SDK, you can migrate `_select_checkpoint()` here as you've done now, because then there's only one copy of it.

I don't feel too strongly about this, so you don't need to treat this comment as blocking.

Reply:
> also, are you basically just copy/pasting the .select_checkpoint() verification logic here?

Yes.

Thanks for thinking about it. This is nice. I like this strategy, too, minus two factors:
1. I want the stack trace to come from the CLI
2. I prefer the CLI errors to mention flags (`--smaller-is-better`) instead of variables (`smaller_is_better`)

What I've chosen is worse from a tech-debt perspective, but I can't think of anything for (2) especially that localizes variable names to library-style or cli-style.